src/sentry/tasks/reports.py

```diff
@@ -211,6 +213,39 @@ def get_aggregate_value(start, stop):
     ]
 
 
+def chunk_event_counts(func):
+    """
+    Chunks queries for getting event counts
+
+    We are chunking up the issue ids and querying because Snuba produces SQL for Clickhouse.
+    There is a max limit on how long the SQL can be. We cross this limit for large accounts
+    and as a result, they cannot produce the weekly report.
+
+    So, with this decorator, we will chunk up the issue IDs and ask for the issue count in those chunks.
+    Then, we will stitch the responses back together.
+    """
+
+    def chunks(arr, chunk_size):
+        for i in range(0, len(arr), chunk_size):
+            yield arr[i : i + chunk_size]
+
+    def wrapper(issue_ids, start, stop, rollup):
+        combined = {}
+
+        for chunk in chunks(list(issue_ids), BATCH_SIZE):
+            single = func(chunk, start, stop, rollup)
+            combined.update(single)
+
+        return combined
+
+    return wrapper
+
+
+@chunk_event_counts
+def get_event_counts(issue_ids, start, stop, rollup):
+    return tsdb.get_sums(tsdb.models.group, issue_ids, start, stop, rollup=rollup)
```

Comment:
The implementation of the `chunk_event_counts` decorator basically couples it to the `get_event_counts` function signature and reduces it's utility as a general purpose, reusable decorator. (In my opinion, single purpose decorators are a bit of a smell — the `snuba` codebase is a good — or bad, depending on how you look at it — example of this, with things like `split_query` that only actually work on one method in practice.)

I think it might be worth it to reconsider this the structure of this function so that it makes it more apparent how the result is being constructed.

```python
def get_event_counts(issue_ids, start, stop, rollup):
        combined = {}
        for chunk in chunked(issue_ids, BATCH_SIZE):
            combined.update(tsdb.get_sums(tsdb.models.group, chunk, start, stop, rollup=rollup))
        return combined
```